### ISSUE TITLE
Question training own dataset

### ISSUE THREAD

**Issue by RJVisee44** [experiencor/basic-yolo-keras#228]
So I am trying to train a relatively large dataset of over 160,000 images for hand detection. I think I understand the code for the most part. Running in Spyder, tensorflow version 1.6.0 and keras version 2.1.3. However, I have a few questions:

1) How does the input size affect the performance? I have images that are 1920x1080. Or 720x405 for the lower resolution images.  

2) Do you expect generate anchors to produce anchors that begin below 1 and generally get larger? My anchors begin above 1: [1.24,1.52, 1.54,2.80, 2.10,4.03, 2.79,2.91, 2.84,5.20]

3) My reported loss per epoch is very low: ~0.03 but my lowest validation loss seems to hang around the 2.1 mark. I am using a very small batch size due to memory, so this could be the reason. Also, my train images:validation images ratio is a little high (about 15:1). Could these result in the poor performance? (mAP is 20%).

4) We can use the full_yolo_backend.h5 weights here for backend, correct? 

**Config File**
```
{
    "model" : {
        "backend":              "Full Yolo",
        "input_size":           416,
        "anchors":              [1.24,1.52, 1.54,2.80, 2.10,4.03, 2.79,2.91, 2.84,5.20],
        "max_box_per_image":    10,        
        "labels":               ["L","R"]
    },

    "train": {
        "train_image_folder":   "M:/basic-yolo-keras-master/Hand_Data/train_3/",
        "train_annot_folder":   "M:/basic-yolo-keras-master/Hand_Data/train_3ann/",     
          
        "train_times":          1,
        "pretrained_weights":   "",
        "batch_size":           4,
        "learning_rate":        1e-4,
        "nb_epochs":            50,
        "warmup_epochs":        0,

        "object_scale":         5.0 ,
        "no_object_scale":      1.0,
        "coord_scale":          1.0,
        "class_scale":          1.0,

        "saved_weights_name":   "full_yolo_hand.h5",
        "debug":                true
    },

    "valid": {
        "valid_image_folder":   "M:/basic-yolo-keras-master/Hand_Data/val/",
        "valid_annot_folder":   "M:/basic-yolo-keras-master/Hand_Data/val_ann/",

        "valid_times":          1
    }
}
```

**Comment by experiencor** [comment]
@RyanCodes44 

1. From my experience, image size does not matter. They will all be resized to input_size.
2. It does not have to be the case. So I see that your objects' sizes do vary too much.
3. There are many things that contribute to the low mAP. Batch size may affect batch norm statistics and hence accuracy. You also need to tweak the 4 scales to improve accuracy.
4. Yes. That file is supposed to be auto loaded into full yolo backend during model creation.

**Comment by RyanCodes44** [comment]
@experiencor Thank you for the reply. 

2) Is it then difficult to use anchors in this case due to the large variations in object size? 

3) Is there any easy way to explain those 4 scales other than what is found in the README? For example, would it be better to increase the penalties? 

If I change these parameters, I assume I have to retrain the entire model? (i.e. start with nothing in "pretrained weights")

**Comment by experiencor** [comment]
2. Sorry, my mistake. I mean your anchors do not vary much.
3. You can train until you have good recall. Don't have to start from scratch but the one you have already trained.

**Comment by RyanCodes44** [comment]
@experiencor Thank you for your response.

One last question surrounding the returned boxes when predicted. Is the only box returned for the class the one with the highest confidence?

**Comment by experiencor** [comment]
Yes, highest confidence among the ones with confidences higher than 0.5

**Comment by jaideep11061982** [comment]
hi i have following questions about the 
 train_imgs, train_labels = parse_annotation(config['train']['train_annot_folder'], 
                                                config['train']['train_image_folder'], 
                                                config['model']['labels'])
1) What is returned in train_imgs ,i tried following parse but coudnt understand much

2) After parse how does the train labels look like can you please give any example. .Since there are no anotations data given it bit hard to follow

3) what is the purpose of overlap after this
4) My objective is to identify only single class which is if an image contains Date or not,if yes where it is...
can we use this model...

**Comment by experiencor** [comment]
1. the image path and associated objects
2. it's just all the labels appeared in the dataset
3. check if you want to detect some class which is not present in the dataset. The code exits if it does.
4. yes, I saw people detecting brand name like Coca Cola with this method

**Comment by mohamedabdallah1996** [comment]
Hi @experiencor , I am working on logos dataset and I want to detect logo in the image and classify this logo. I have 32 class (logos). 
How can I choose the appropriate "pretrained_weights" or I should train from scratch?
